internal/extension/types/manager.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/envoyproxy/gateway/api/config/v1alpha1"
+)
+
+// Extension represents an extension registered by the Manager.
+type Extension struct {
+	v1alpha1.Extension
+
+	SupportedHookTypes []HookType
+}
+
+// Manager handles and maintains registered extensions and returns clients for
+// different Hook types.
+type Manager interface {
+	// HasExtension checks to see whether a given Group and Kind has an
+	// associated extension registered for it.
+	//
+	// If a Group and Kind is registered with an extension, then it should
+	// return true and the id of the extension, otherwise return
+	// (false, nil).
+	HasExtension(g v1beta1.Group, k v1beta1.Kind) (bool *string)
+
+	// GetExtension returns an Extension for the provided extension id.
+	//
+	// If no Extension exists for that id, then it should return (nil, error).
+	GetExtension(eid string) (*Extension, error)
+
+	// ListExtensions returns a list of all extensions registered by the Manager.
+	ListExtensions() []*Extension
```

Comment:
Same on this one, The only use case I can think of for exposing this is for logging or is there any reason to have this?

Reply:
see above comment